src/libs/utils.js

```diff
@@ -382,6 +382,55 @@ export const useConsoleAssert = (condition, message) => {
   }
 }
 
+/**
+ * Asserts that a component has an accessible label, and alerts the developer how to fix it if it doesn't.
+ *
+ * @param componentName The name of the component, which will be printed to the console if there's an alert.
+ * @param [allowLabelledBy] If true (default), the component can have an aria-labelledby linked to another element. Set to false to only allow aria-label.
+ * @param [allowId] If true, the component can have an id linked to a label using htmlFor. This is true for form elements.
+ * @param [allowTooltip] If true, the component can have a tooltip which will be used if needed as the label.
+ * @param [allowContent] If true, the component can used nested textual content as its label, as long as it's not a single unlabelled icon
+ * @param [ariaLabel] Optional: The label provided to the component
+ * @param [ariaLabelledBy] Optional: The ID of the label provided to the component
+ * @param [id]: Optional: The ID of the component if allowId is true
+ * @param [tooltip] Optional: The tooltip provided to the component if allowTooltip is true
+ */
+export const useLabelAssert = (componentName, {
+  allowLabelledBy = true, allowId = false, allowTooltip = false, allowContent = false,
+  'aria-label': ariaLabel, 'aria-labelledby': ariaLabelledBy, id, tooltip
+}) => {
+  const printed = useRef(false)
+
+  if (!printed.current) {
+    // Ensure that the properties contain a label
+    if (!(ariaLabel ||
+      (allowLabelledBy && ariaLabelledBy) ||
+      (allowId && id) || (allowTooltip && tooltip)
+    )) {
+      printed.current = true
+
+      console.warn(`For accessibility, ${componentName} needs a label. Resolve this by doing any of the following: ${allowContent ? `
+  * add a child component with textual content or a label
+  * if the child is an icon, add a label to it` : ''}${allowTooltip ? `
+  * add a tooltip property to this component, which will also be used as the aria-label` : ''}
+  * add an aria-label property to this component${allowLabelledBy ? `
+  * add an aria-labelledby property referencing the id of another component containing the label` : ''}${allowId ? `
+  * create a label component and point its htmlFor property to this component's id` : ''}`)
+    }
+  }
+}
+
+/**
+ * Returns the aria-label to use for a component if applicable.
+ *
+ * If aria-label was provided, it will be returned
+ * If aria-labelledby was provided, or an id was provided and allowId is true, then this will return nothing because the component is already labeled
+ * If a tooltip was provided without any of the other attributes, it will be returned to be used as the aria-label
+ */
+export const useAriaLabelOrTooltip = ({ allowId = false, 'aria-label': ariaLabel, 'aria-labelledby': ariaLabelledBy, id, tooltip }) => {
```

Comment:
Could this be renamed to `getAriaLabelOrTooltip`.
The reason being this isn't a react hook and is being used deeper in the render function (rather than at the top) to provide values to the props of other components. 

The `use` in the name could confuse developers